### ISSUE TITLE
Exposing hooks for renovate

### ISSUE THREAD

**Issue by or-adar** [renovatebot/renovate#14604]
### What would you like Renovate to be able to do?

Hi,
I'm currently running renovate on a self hosted GitLab Runner 
and I would love renovate to expose a hook that will allow me to run some custom scripts
only if it updated all dependencies and after all relevant pull requests were created/updated.

My use case for this currently is that my team manages a lot of repos and we want to run a custom script that eventually will send a custom email to relevant members that will include a customize variant of the "Dependency Dashboard", and having a hook that will trigger this script whenever renovate updates or creates all relevant PRs will make our task much simpler.

Also,
I do know that GitLab sends emails for events on Issues & MR, but we don't want to rely on this behaviour, as we get a lot of noise from GitLab and this is less of a maintainable option for us.

If renovate already supports this behaviour, I would like to know what I need to define so renovate will trigger my script on the relevant events.

### If you have any ideas on how this should be implemented, please tell us here.

exposing a field for renovate's config file that will trigger the script/s passed.

### Is this a feature you are interested in implementing yourself?

No

**Comment by viceice** [comment]
What about post upgrade tasks?

**Comment by rarkins** [comment]
It sounds like you're asking for a "post run" hook, rather than per-branch/PR?

**Comment by or-adar** [comment]
@rarkins Correct! but only if renovate updated the Dependency Dashboard issue

@viceice I read about it, however that hook runs per branch and before renovate creates a commit (or even creates/updates PRs) which is less ideal for me, as I'll want to get the latest content of "Dependency Dashboard" issue (which renovate maintains).
